ecosystem/sep-0010.md

```diff
@@ -94,21 +96,19 @@ On success the endpoint must return `200 OK` HTTP status code and a JSON object
   * source account set to server's signing account
   * invalid sequence number (set to 0) so the transaction cannot be run on the Stellar network
   * time bounds: `{min: now(), max: now() + 900 }` (we recommend expiration of 15 minutes to give user time to sign transaction)
-  * operations: `manage_data(source: client_account, key: '<anchor name> auth', value: random_nonce())`
-    * The value of key is not important, but can be the name of the anchor followed by `auth`. It can be at most 64 bytes.
-    * The value must be 64 bytes long. It contains a 48 byte cryptographic-quality random string encoded using base64 (for a total of 64 bytes after encoding).
-  * signature by the web service signing account
+  * operations: `manage_data(source: client_account, key: '<anchor name> auth', value: '<home domain>')`
```

Comment:
I don't think we should put home domain in place of the random nonce. First, the cryptographic-quality random nonce is important to keep challenge transactions unique for every request, without it challenge becomes deterministic and susceptible to different attacks. Strictly speaking, for proper mutual authentication there should be also a client nonce, which is originally sent to the server and is included in the transaction payload.

Another thing I want to point out, is that `key` of the manage data operation is defined in XDR as 64-byte string and is considered human-readable, while the value is 64-byte opaque binary value and mostly represented in  base64-encoded form. Placing home domain into the key will make it possible to visually verify the payload, for example before signing on a hardware wallet. I would suggest using key of `<home domain> auth` and keeping the value as a random nonce.

Reply:
I agree, adjusted.